ci/infra/openstack/generate-cloud-provider-conf

```diff
@@ -0,0 +1,72 @@
+#!/bin/bash
+
+if [ -z "${OS_AUTH_URL}" ] || [ -z "${OS_USERNAME}" ] || \
+   [ -z "${OS_PASSWORD}" ] || [ -z "${OS_PROJECT_ID}" ] || \
+   [ -z "${OS_PRIVATE_SUBNET_ID}" ] || [ -z "${OS_PUBLIC_NET_ID}" ]; then
+  echo '[ ERROR ] $OS_AUTH_URL $OS_USERNAME $OS_PASSWORD $OS_PROJECT_ID'
+  echo '[ ERROR ] $OS_PRIVATE_SUBNET_ID $OS_PUBLIC_NET_ID must be specified'
+  echo '[ ERROR ] Please download and source your OpenStack RC file'
+  exit 1
+fi
+
+OPENSTACK_CONF="openstack.conf"
+
+umask 077
+
+cat << EOF > $OPENSTACK_CONF
+[Global]
+auth-url="${OS_AUTH_URL}"
+username="${OS_USERNAME}"
+password="${OS_PASSWORD}"
+tenant-id="${OS_PROJECT_ID}"
+tenant-name="${OS_PROJECT_NAME}"
+domain-id="${OS_USER_DOMAIN_ID}"
+domain-name="${OS_USER_DOMAIN_NAME}"
+region="${OS_REGION_NAME}"
+ca-file="${CA_FILE}"
+[LoadBalancer]
+lb-version=v2
+subnet-id="${OS_PRIVATE_SUBNET_ID}"
+floating-network-id="${OS_PUBLIC_NET_ID}"
+create-monitor=yes
+monitor-delay=1m
+monitor-timeout=30s
+monitor-max-retries=3
+[BlockStorage]
+trust-device-path=false
+bs-version=v2
+ignore-volume-az=true
+EOF
+
+umask 022
+
+[ -z "$OS_PROJECT_NAME" ] && sed -i '/^tenant-name=/d' $OPENSTACK_CONF
+[ -z "$OS_USER_DOMAIN_ID" ] &&  sed -i '/^domain-id=/d' $OPENSTACK_CONF
+[ -z "$OS_USER_DOMAIN_NAME" ] && sed -i '/^domain-name=/d' $OPENSTACK_CONF
+[ -z "$CA_FILE" ] && sed -i '/^ca-file=/d' $OPENSTACK_CONF
+
+if [ -z "${TR_STACK}" ] || [ -z "${TR_LB_IP}" ] || [ -z "$TR_MASTER_IPS" ] || [ -z "$TR_WORKER_IPS" ]; then
+  echo '[ ERROR ] $TR_STACK $TR_LB_IP $TR_MASTER_IPS $TR_WORKER_IPS must be specified'
+  exit 1
+fi
+
+echo -e "\n[ INFO ] ### Run following commands to bootstrap skuba cluster:\n"
+echo " skuba cluster init --control-plane ${TR_LB_IP} --cloud-provider openstack ${TR_STACK}-cluster"
+echo " mv openstack.conf ${TR_STACK}-cluster/cloud/openstack.conf"
+echo " cd ${TR_STACK}-cluster"
+
+MASTERS=($TR_MASTER_IPS)
+echo " skuba node bootstrap --target ${MASTERS[0]} --sudo --user sles caasp-master-${TR_STACK}-0"
+if [ "${#MASTERS[@]}" -gt "1" ]; then
+  i=0
+  for MASTER in $TR_MASTER_IPS; do
+    if [ $i -eq "0" ]; then ((++i)); continue; fi
```

Comment:
Why not echo the bootstrap command here in place of the `continue`, and make this an if/else? Then the array will also be unnecessary. Some readers get confused by shell arrays. :) 

Either way, the increment of `i` also needs to happen outside the if, which would be easier with an `else` rather than `continue`. :)

Reply:
Good idea :+1: